File: algorithm/basic/소인수분해.py
def solution(n):

    prime_numbers = []
    i = 2
    while i<=n:
        if n % i == 0:
            prime_numbers.append(i)
            n = n // i # '몫'
        else :
            i = i+1

    return list(dict.fromkeys(prime_numbers))


print(solution(12))

File: algorithm/basic/javascript/소인수분해.js
const n = 12;

function solution(n) {
    const prime_factors = [];
    for (let i = 2; i < n + 1; i++) {
        while (n % i === 0) {
            n = n / i;
            if (n % i != 0) prime_factors.push(i);
        }
    }
    return prime_factors;
}

console.log(solution(n));

// 소인수 : Prime Factor
